pkg/products/monitoring/prometheusRules.go

```diff
@@ -292,6 +292,22 @@ func (r *Reconciler) newAlertsReconciler() resources.AlertReconciler {
 					},
 				},
 			},
+			{
+				AlertName: "install-upgrade-alerts",
+				Namespace: r.Config.GetOperatorNamespace(),
+				GroupName: "general.rules",
+				Rules: []monitoringv1.Rule{
+					{
+						Alert: "RHMICSVAbnormal",
+						Annotations: map[string]string{
+							"message": "An abnormal CSV was found, possibly preventing an operator from installing/upgrading",
+						},
+						Expr:   intstr.FromString(fmt.Sprintf("csv_abnormal{reason='RequirementsNotMet',phase='Pending'} or csv_abnormal{reason='RequirementsNotMet',phase='Failed'}")),
```

Comment:
Should we consider both Pending and Failing, or should this be just for Failing CSVs?

Reply:
It's mentioned in the in the Jira description that operators most often get stuck in either phase, but I'm not sure how this translates to the RHMI operator, and I'm not that familiar with upgrades TBH.

If sticking with both Pending and Failed, could the expression be shortened to something like: `csv_abnormal{reason='RequirementsNotMet',phase='Pending|Failed'}` I wonder?

Reply:
`phase=~'Pending|Failed'` should do the trick

Reply:
Do we need to limit the expression to RHMI related namespaces?

Reply:
Do we need to limit the expression to `RHMI` related namespaces?

Reply:
Do we need to limit the expression to `RHMI` related namespaces?

Reply:
I'll try the shortened version again, I had it in before but it wasn't giving me the results I expected when testing, I might have been missing the `~` though, which might be the culprit, I'll also limit the namespaces to `rhmi`, the cluster prometheus already has a similar query that watches all (redhat)namespaces, also we don't want to get alerts for non-managed products so it makes sense to limit it to our namespaces